### ISSUE TITLE
Glyph Atlas isn't cleaned up properly

### ISSUE THREAD

**Issue by kkaefer** [mapbox/mapbox-gl-native#1444]
![mapbox gl gputrace 2015-05-06 17-45-39](https://cloud.githubusercontent.com/assets/52399/7496951/ba3de6d4-f417-11e4-80e3-9561ddf16d7f.png)

**Comment by mikemorris** [comment]
Possible this is causing https://github.com/mapbox/node-mapbox-gl-native/issues/106 as well?

**Comment by friedbunny** [comment]
#3781 should improve this. #1681 is likely caused by this issue.

**Comment by jfirebaugh** [comment]
Should be fixed by #8341.

**Comment by ChrisLoer** [comment]
Are we just hoping #8341 will fix this by shaking things up? There's a new implementation, but the tile reference counting scheme is still essentially the same.

**Comment by jfirebaugh** [comment]
I think we should check if this issue still exists after #8341 lands. It shakes things up enough that it seems like there's a pretty good chance any remaining issue would in fact be an issue in #8341 itself.

**Comment by ChrisLoer** [comment]
Do we have a test protocol that exposes this behavior? #8341 is in now, so this is speculatively fixed. As a simple test, I loaded a bunch of chinese tiles with the glyph atlas debug view, watched the atlas fill up with Chinese glyphs, then panned over to France, zoomed in, and watched them all get cleaned out as the tiles got pushed out of the cache. However, that's the same behavior I saw before my changes, so I'm not sure it's a useful test.
